CIF-652 - CIF Cloud support for viewing product description within Product Console

dplaton opened the pull request:
## Description 

A little note on the scaffolding feature, as it's used in Commerce - the Products console has two ways of checking the scaffolding:
- using the `cq:scaffolding` property of the resource, which should point to a valid scaffolding
- searching for the scaffolding in `/apps` and find the one which has a `cq:targetPath` property pointing at the root of the catalog.

To make the cloud products use a scaffolding we had two options:
- create a sample scaffolding anywhere and inject the `cq:scaffolding` property at runtime in the synthetic resources. This approach is not ideal since it mixes our Resource API implementation with the UI domain (a scaffolding is an UI concept)
- create a sample scaffolding in `/apps`, but then the user has to update it manually to point it to the root of the virtual catalog. This step is necessary because we cannot know the root of the catalog since the user defines it using the Bind Products Wizard.

We choose to go with the latter approach and document the steps that the users have to perform.

## Related Issue

CIF-652

## Motivation and Context

Provide a generic scaffolding so that the users can see the properties of a cloud product.

## How Has This Been Tested?

Functional testing

## Screenshots (if appropriate):

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

dplaton committed:
CIF-652 - CIF Cloud support for viewing product description within Product Console

A little note on the scaffolding feature, as it's used in Commerce - the Products console has two ways of checking the scaffolding:
- using the `cq:scaffolding` property of the resource, which should point to a valid scaffolding
- searching for the scaffolding in `/apps` and find the one which has a `cq:targetPath` property pointing at the root of the catalog.

To make the cloud products use a scaffolding we had two options:
- create a sample scaffolding anywhere and inject the `cq:scaffolding` property at runtime in the synthetic resources. This approach is not ideal since it mixes our Resource API implementation with the UI domain (a scaffolding is an UI concept)
- create a sample scaffolding in `/apps`, but then the user has to update it manually to point it to the root of the virtual catalog. This step is necessary because we cannot know the root of the catalog since the user defines it using the Bind Products Wizard.

We choose to go with the latter approach and document the steps that the users have to perform.

dplaton committed:
Merge branch 'master' into issue/CIF-652

herzog31 commented:
@dplaton Changes look good to me. Can you additionally add an IT like https://github.com/adobe/commerce-cif-connector/blob/master/it/http/src/test/java/com/adobe/cq/commerce/it/http/GraphqlProductConsoleIT.java#L219-L236 to the PR?

dplaton commented:
> Can you additionally add an IT ...

Already working on it

dplaton committed:
CIF-652 - CIF Cloud support for viewing product description within Product Console

Update the scaffolding to display the SKU instead of the id.
Add an IT to check that the product properties page contains the minimal set of data.

dplaton committed:
CIF-652 - CIF Cloud support for viewing product description within Product Console

Fix the formatting.
